zcash_script/src/descriptor.rs

```diff
@@ -0,0 +1,345 @@
+//! Output Script Descriptors for Zcash.
+//!
+//! This module provides the subset of [BIP 380] Output Script Descriptors that are valid
+//! to use within the Zcash ecosystem.
+//!
+//! [BIP 380]: https://github.com/bitcoin/bips/blob/master/bip-0380.
+
+use alloc::vec::Vec;
+use core::fmt;
+use core::iter;
+
+use bip32::{ExtendedPublicKey, Prefix};
+use ripemd::Ripemd160;
+use secp256k1::PublicKey;
+use sha2::{Digest, Sha256};
+use thiserror::Error;
+
+use crate::{
+    op,
+    script::{self, Evaluable},
+};
+
+mod encoding;
+pub use encoding::ParseError;
+
+/// A key expression.
```

Comment:
Should some of this be in [[`zcash_transparent::keys`](https://github.com/zcash/librustzcash/blob/main/zcash_transparent/src/keys.rs)?

Reply:
There's a layer problem here: `zcash_transparent` depends on `zcash_script`, but the cescriptor logic (which depends on key expressions) definitely belongs in `zcash_script`. So I've tried to keep this module pretty close to what BIP 380 and BIP 388 specify, without any Zcash-specific affordances (other than restricting the impl to the relevant subset).

The `zcash_transparent::keys` equivalent for ZIP 48 is in https://github.com/zcash/librustzcash/pull/1962